test/js/node/http/node-http.test.ts

```diff
@@ -1746,3 +1746,37 @@ if (process.platform !== "win32") {
     expect([joinPath(import.meta.dir, "node-http-ref-fixture.js")]).toRun();
   });
 }
+
+it("#10177 response.write with non-ascii latin1 should not cause duplicated character or segfault", async () => {
+  const expected = "á" + "-".repeat(254) + "x";
```

Comment:
Please add a test that checks for N variations of sizes
```suggestion
for (let encodedChar of ["á", "📙", "👍🏽"] {
for (let repeatCount = 0; repeatCount < 512; repeatCount++) {
  const expected = encodedChar + "-".repeat(repeatCount) + "x";
}
...
```

Reply:
The reason for 👍🏽 is its a grapheme of 👍 🟤  

The reason for `á` is its a latin1 supplemental character

Reply:
variations with some relevant sizes https://github.com/oven-sh/bun/pull/10221/commits/c7afd02ad7c4fe70669c3b07fc899aa4a5281990